--- a/src/dispatch.c
+++ b/src/dispatch.c
@@ -490,7 +490,8 @@
             if (gpu->glsl.vulkan) {
                 ADD(pre, "layout(binding=%d) uniform %c%s "$";\n",
                     desc->binding, prefix, type, id);
-            } else if (gpu->glsl.gles && prefix != ' ') {
+            } else if (gpu->glsl.gles &&
+                       (prefix != ' ' || tex->params.format->component_depth[0] > 8)) {
                 ADD(pre, "uniform highp %c%s "$";\n", prefix, type, id);
             } else {
                 ADD(pre, "uniform %c%s "$";\n", prefix, type, id);
